src/components/Account-panel/Order-history/OrdersList.tsx

```diff
@@ -0,0 +1,197 @@
+import {
+  Dispatch,
+  FC,
+  Fragment,
+  SetStateAction,
+  useCallback,
+  useEffect,
+  useRef
+} from 'react'
+import { Box, Divider } from '@mui/material'
+import { format } from 'date-fns'
+import { uk } from 'date-fns/locale'
+import { Typography } from '../../UI/Typography'
+import { Status, StepCustom } from '../../Step/Step'
+import { useTheme } from '@mui/material/styles'
+
+export type OrdersType = {
+  ordered_products: any[]
+  created_at: string
+  status_order: Status
+  price_order: number
+  id: number
+}
+
+export type SelectedOrder = {
+  id: number
+  ordered_products: number
+}
+
+type OrdersListProps = {
+  orders: OrdersType[]
+  selectedOrderId: SelectedOrder | null
+  loading: boolean
+  hasMore: boolean
+  setSelectedOrderId: Dispatch<SetStateAction<SelectedOrder | null>>
+  setPage: Dispatch<SetStateAction<number>>
+  setSelectedOrderProducts: Dispatch<SetStateAction<any[]>>
+}
+
+const formatDate = (dateString: string) => {
+  return format(new Date(dateString), 'd MMM yyyy', { locale: uk })
+}
+
+export const getProductLabel = (count: number) => {
+  if (count === 1) return 'товар'
+  if (count >= 2 && count <= 4) return 'товари'
+  return 'товарів'
+}
+
+export const OrdersList: FC<OrdersListProps> = ({
+  orders,
+  setSelectedOrderProducts,
+  selectedOrderId,
+  setSelectedOrderId,
+  loading,
+  hasMore,
+  setPage
+}) => {
+  const theme = useTheme()
+
+  const observer = useRef<IntersectionObserver>()
+
+  const lastOrderElementRef = useCallback(
+    (node: Element | null) => {
+      if (loading) return
+      if (observer.current) observer.current.disconnect()
+      observer.current = new IntersectionObserver((entries) => {
+        if (entries[0].isIntersecting && hasMore) {
+          setPage((prevPage: number) => prevPage + 1)
+        }
+      })
+      if (node) observer.current.observe(node)
+    },
+    [loading, hasMore]
+  )
+
+  useEffect(() => {
+    const selectedOrder = orders.find(
+      (order) => order.id === selectedOrderId?.id
+    )
+    if (selectedOrder) {
+      setSelectedOrderProducts(selectedOrder.ordered_products)
+    }
+  }, [selectedOrderId])
+
+  const handleOrderClick = (orderId: number) => {
+    const selectedOrder = orders.find((order) => order.id === orderId)
+    if (selectedOrder) {
+      setSelectedOrderId({
+        id: orderId,
+        ordered_products: selectedOrder.ordered_products.length
+      })
+      setSelectedOrderProducts(selectedOrder.ordered_products)
+    }
+  }
+
+  const renderOrder = (
```

Comment:
do not do `render_something` function. this should be a component

Reply:
done